### ISSUE TITLE
binary release still not compiled for 10.11+

### ISSUE THREAD

**Issue by tessus** [horosproject/horos#191]
I've just noticed that the binary release is still only available for 10.12, which makes really no sense.
You only have to make the changes I've mentioned in https://github.com/horosproject/horos/issues/171#issuecomment-287413303

In any case, I've created a binary for 10.11+ for anyone who needs it:
https://evermeet.cx/pub/macosx/Horos/Horos-2.2.0.zip

**Comment by boskar** [comment]
Hello, do You think it is possible to backport current horos (3.0 beta) to 10.11?

**Comment by tessus** [comment]
I don't think there's a need to backport any code. You only have to make the changes in the project I described in https://github.com/horosproject/horos/issues/171#issuecomment-287413303

**Comment by fvpolpeta** [comment]
Thanks for your advice guys.

This will reviewed before the final release of version 3.0.

We get in touch soon again.

Thanks

Fauze

**Comment by tessus** [comment]
@boskar when I pulled the latest code, I saw that the deployment target was actually set to 10.11.

Thus, the version which is available for download should run on 10.11, unless the release team compiled a version which is not the same as the code on github.

However, I also compiled a version (not tested though) and put it here:
https://evermeet.cx/pub/macosx/Horos/Horos-3.0.0.rc1.zip

@fvpolpeta after the reorganization of the source tree and project, there are 2 compile errors, because paths are not set correctly by the scripts and Xcode. `cmake` and `pkg-config` cannot be found. Xcode **does not** use the user's environment PATH. cmake and pkg-config are not part of macOS, thus they are either in /opt/local/bin (macports) or some other directory for homebrew. In any case, I had to create a symbolic link in /bin for those 2 binaries.
Maybe you should use the README.md file to describe how to compile the project. Especially since the project does not compile out of the box. (also a lot of people don't know how to use git submodules)

**Comment by fvpolpeta** [comment]
Thanks for your feedback. We will be writing building instructions when we conclude the release of version 3.0.

Some pieces are recent and we are still accommodating them. In advance, if you feel comfortable on consolidating the README.md with build instructions and make a pull request, we will be very pleased to accept it and incrementally improve it. That’s the spirit of open source. 

Regarding the 10.11 version, we will be investigating that as mentioned. Perhaps, it’s something related to the signing process/archiving of XCode. We will see.

Thanks again.

Fauze

**Comment by tessus** [comment]
@fvpolpeta 

> README.md with build instructions and make a pull request, we will be very pleased to accept it and incrementally improve it. 

Yep, sounds good. Will do. Not sure when I will be able to get around to it though, maybe this weekend.

> Regarding the 10.11 version, we will be investigating that as mentioned. Perhaps, it’s something related to the signing process/archiving of XCode. We will see. 

I'm not sure what you mean by that. You've changed the deployment target already, so there's nothing else you have to do. If you compile the current master, it will run on 10.11 and later.

**Comment by horosproject** [comment]
We haven’t changed the target to anything different than 10.11.

Since we do generate and sign the App via Archive, I will investigate if that may be imposing some change. Not sure.

I will let you know. As said, the idea is to have this resolved before we make the 3.0 the official release.

Cheers

**Comment by tessus** [comment]
When I opened this issue, the minimum deployment target in the project was 10.12.

Now it is not. So you must have changed it from 10.12 to 10.11. ;-)

**Comment by fvpolpeta** [comment]
Oh yeah. It was changed time ago based on the issue you reported. But there are still users complaining about this (not 10.11 compliant).

I will be reviewing all targets and dependencies (and the signing process).

Your help is appreciated.

Thanks

**Comment by tessus** [comment]
Ah, I didn't know that people still had an issue with 10.11. It compiles perfectly (there are a few deprecation warnings for NSForm), but other than that this project has no warnings - which is remarkable btw.

I've already upgraded to High Sierra, but I still have a backup of my 10.11 on an external disk, which I can boot up at any time.
(I compile other stuff for 10.9 and above, so that's not really an issue.)

You can always add me to a ticket by mentioning me. e.g.: /cc @tessus
In this case I can follow the 10.11 issues...

**Comment by fvpolpeta** [comment]
@tessus sorry, I didn’t explain myself properly. The issue I mean is the fact I have been building for 10.11 but the binary version built and signed requires recent versions of macOS.

**Comment by tessus** [comment]
Hmm, this makes no sense. If you build for 10.11 it has to run on 10.11, otehrwise there's a bug in Xcode. Signing should be OS version agnostic. Very strange, I hope you can figure out what the problem is.

**Comment by tessus** [comment]
@fvpolpeta 

> README.md with build instructions and make a pull request, we will be very pleased to accept it and incrementally improve it.

please see #273
